default/python/AI/CombatRatingsAI.py

```diff
@@ -148,7 +149,7 @@ def __get_stats_from_ship(self):
                         # TODO: Depending on future implementation, might actually need to handle this case.
                         warn("Multiple hangar types present on one ship, estimates expected to be wrong.")
                     fighter_damage = max(fighter_damage, part_damage)
-        self._basic_stats = self.BasicStats(attacks, structure, shields)
+        self._basic_stats = self.BasicStats(attacks, int(structure), int(shields))
```

Comment:
should fix these in the constructor to be ```float``` instead considering they are floats in the game engine.

Reply:
It is documented as int, so class users expect to get int here.   But I don't have any strong preferences here.

Reply:
int is a bug.

Reply:
I have already pushed commit with the fix